features/all-positions/LiquidationActionDialog.tsx

```diff
@@ -0,0 +1,489 @@
+  Typography,
+  Dialog,
+  Button,
+  Tooltip,
+} from "@material-ui/core";
+
+import EmpState from "../../containers/EmpState";
+import DvmState from "../../containers/DvmState";
+import Token from "../../containers/Token";
+import EmpSponsors from "../../containers/EmpSponsors";
+import EmpContract from "../../containers/EmpContract";
+import Collateral from "../../containers/Collateral";
+import Etherscan from "../../containers/Etherscan";
+
+import { DOCS_MAP } from "../../utils/getDocLinks";
+import { isPricefeedInvertedFromTokenSymbol } from "../../utils/getOffchainPrice";
+
+const Label = styled.span`
+  color: #999999;
+`;
+
+const Status = styled(Typography)`
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+const PositionDialog = styled.div`
+  width: 100%;
+  border-color: white;
+  padding-left: 15px;
+  padding-right: 15px;
+  padding-bottom: 20px;
+`;
+
+const Important = styled(Typography)`
+  color: red;
+  background: black;
+  display: inline-block;
+`;
+
+const Link = styled.a`
+  color: white;
+  font-size: 14px;
+`;
+
+interface DialogProps {
+  handleClose(): any;
+  isDialogShowing: boolean;
+  selectedSponsor: string | null;
+}
+
+const LiquidationActionDialog = (props: DialogProps) => {
+  const { empState } = EmpState.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { finalFee } = dvmState;
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { contract: emp } = EmpContract.useContainer();
+
+  const { symbol: tokenSymbol } = Token.useContainer();
+  const {
+    symbol: collSymbol,
+    balance: collBalance,
+    allowance: collAllowance,
+    setMaxAllowance: setMaxCollateralAllowance,
+  } = Collateral.useContainer();
+  const { liquidations } = EmpSponsors.useContainer();
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    disputeBondPct,
+    priceIdentifier: priceId,
+    liquidationLiveness,
+    withdrawalLiveness,
+    currentTime,
+    isExpired,
+  } = empState;
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String) => {
+    return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+  };
+
+  if (
+    emp !== null &&
+    liquidations !== null &&
+    props.selectedSponsor !== null &&
+    liquidations[props.selectedSponsor] &&
+    liquidations[props.selectedSponsor] !== null &&
+    isExpired !== null &&
+    priceId !== null &&
+    disputeBondPct !== null &&
+    collAllowance !== null &&
+    collBalance !== null &&
+    collSymbol !== null &&
+    liquidationLiveness !== null &&
+    withdrawalLiveness !== null &&
+    currentTime !== null &&
+    tokenSymbol !== null &&
+    finalFee !== null
+  ) {
+    const liquidatedPosition = liquidations[props.selectedSponsor];
```

Comment:
Have you tested this with my latest changes to `master`? The key for `liquidations` is now `sponsorPlusId`

Reply:
Nevermind, I'll just assume you make the neccessary changes to `AllLiquidations` and that a sponsor address is passed correctly to here

Reply:
ye, I had not merged master when you left this review comment. PTAL.